apisix/plugins/gzip.lua

```diff
@@ -100,6 +104,14 @@ function _M.check_schema(conf)
 end
 
 
+function _M.rewrite(conf, ctx)
+    if conf.disable_via == true then
+        -- According to nginx source code, via header will cause gzip not work
+        core.request.set_header(ctx, "via", nil)
```

Comment:
If this can satisfy your requirement, why not just use proxy-rewrite plugin to do so?

Reply:
I just think that this is a control that the gzip plugin lacks, and there is no need to introduce additional plugins. (of course, proxy-rewrite can also be implemented)